### PR TITLE
feat(calling): webex-initilaize-changes

### PR DESCRIPTION
**Pull request by Kesari3008**
# COMPLETES #< INSERT LINK TO ISSUE >

## This pull request addresses
We are adding support for developers to be able to use calling SDK without initialized Webex object if they just have access token. They can share the token and calling SDK will initialize Webex object, finish WDM registration and connect with mercury etc. on develop's behalf and return it back to them.
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-422961 

## by making the following changes

Add new method to accept the access token and call Webex.init(). Once Webex is ready, return the initialized Webex object.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [x] Tooling change
- [x] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/calling/calling/jest.expectExtensions.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
 expect.extend({
+  /**
+   * @param received
+   * @param {...any} expected
+   */
```
Comment (BhargavSatya) [inline packages/calling/calling/jest.expectExtensions.js:5]: Please remove or fill the description.

File: packages/calling/calling/jest.expectExtensions.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
   toBeCalledOnceWith(received, ...expected) {
     try {
       expect(received).toBeCalledTimes(1);
       expect(received).toBeCalledWith(...expected);
     } catch (error) {
-      return {message: () => error, pass: false};
+      return {
+        /**
+         *
+         */
```
Comment (BhargavSatya) [inline packages/calling/calling/jest.expectExtensions.js:14]: same as above.

File: packages/calling/calling/src/CallingClient/CallingClient.ts [hunk 3]
```diff
@@ -981,9 +980,9 @@ export class CallingClient extends Eventing<CallingClientEventTypes> implements
           const abort = await this.registerDevice(FAILBACK_UTIL);
 
           if (!abort && !this.isDeviceRegistered()) {
-            const abort = await this.restorePreviousRegistration(FAILBACK_UTIL);
+            const abortNew = await this.restorePreviousRegistration(FAILBACK_UTIL);
```
Comment (BhargavSatya) [inline packages/calling/calling/src/CallingClient/CallingClient.ts:983]: Was there any exception by eslint on this? (It seems fine to have same variable since there scope is in different blocks anyway)

File: packages/calling/calling/versionPlugin.js [hunk 4]
```diff
@@ -27,6 +30,7 @@ const replaceVersion = (err, data) => {
 };
 
 /**
+ * .
```
Comment (BhargavSatya) [inline packages/calling/calling/versionPlugin.js:33]: Would be better to add some descriptions, now that we are public

File: packages/calling/calling/versionPlugin.js [hunk 5]
```diff
@@ -8,9 +8,12 @@ let file;
 
 // eslint-disable-next-line consistent-return
 /**
+ * .
```
Comment (BhargavSatya) [inline packages/calling/calling/versionPlugin.js:11]: Would be better to add some descriptions, now that we are public

File: packages/calling/calling/package.json [hunk 6]
```diff
@@ -43,7 +43,7 @@
     "jest-html-reporters": "3.0.11",
     "platform": "1.3.6",
     "uuid": "8.3.2",
-    "webex": "1.154.1",
+    "webex": "workspace:^",
```
Comment (BhargavSatya) [inline packages/calling/calling/package.json:46]: Pls revert this.

File: packages/calling/calling/package.json [hunk 7]
```diff
@@ -61,8 +61,6 @@
     "@types/mocha": "9.0.0",
     "@types/node": "16.11.9",
     "@types/uuid": "8.3.4",
```
Comment (BhargavSatya) [inline packages/calling/calling/package.json:63]: Why are we removing eslint related plugins from calling. We are also adding calling into .eslintignore. Wouldn't this remove all eslint related checks on calling package?

**Review by BhargavSatya — CHANGES_REQUESTED** [review]
Minor comments. One comment regarding eslint also needs to be resolved.

File: packages/calling/calling/src/CallingClient/CallingClient.ts (hunk 3 above)
Comment (Kesari3008) [reply packages/calling/calling/src/CallingClient/CallingClient.ts:983]: We are getting lint errors for this as per rule : @typescipt-eslint/no-shadow

File: packages/calling/calling/package.json (hunk 6 above)
Comment (Kesari3008) [reply packages/calling/calling/package.json:46]: No without this we won't be able to import webex inside calling which is required for initialization

File: packages/calling/calling/package.json (hunk 7 above)
Comment (Kesari3008) [reply packages/calling/calling/package.json:63]: We have these plugins already available in package.json globally present so calling can also use those. And we have added calling .eslintignore so that rules that are there in global .eslintrc.js file are not being run for calling since we have our own .eslintrc.json file which is still being used and performing eslint related checks for calling package

File: packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts [hunk 8]
```diff
@@ -304,7 +304,7 @@ describe('plugin-dss', () => {
     describe('#_request', () => {
       it('handles a request correctly', async () => {
         webex.request = sinon.stub();
-        uuid.v4.returns('randomid');
+        uuid.v4();
```
Comment (sreenara) [inline packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts:307]: Can we update the copyright information at the head of this file?

File: packages/calling/calling/src/CallingClient/CallingClient.ts [hunk 9]
```diff
@@ -981,9 +980,9 @@ export class CallingClient extends Eventing<CallingClientEventTypes> implements
...
 
-            if (abort) {
+            if (abortNew) {
```
Comment (sreenara) [inline packages/calling/calling/src/CallingClient/CallingClient.ts:985]: Can we rename this something better than abortNew? THe name of the variable doesn't really indicate what's different.

File: packages/calling/calling/src/CallingClient/CallingClient.test.ts [hunk 10]
```diff
@@ -1416,7 +1416,6 @@ describe('CallingClient Tests', () => {
       const restoreSpy = jest.spyOn(callingClient as any, 'restorePreviousRegistration');
       const restartRegSpy = jest.spyOn(callingClient as any, 'restartRegistration');
       const reconnectSpy = jest.spyOn(callingClient as any, 'reconnectOnFailure');
-      const logSpy = jest.spyOn(log, 'log');
```
Comment (sreenara) [inline packages/calling/calling/src/CallingClient/CallingClient.test.ts:1419]: Why has this been removed? I don't see the log removed from the code

**Review by sreenara — CHANGES_REQUESTED** [review]
Please see the comments

File: packages/calling/calling/src/CallingClient/CallingClient.test.ts (hunk 10 above)
Comment (Kesari3008) [reply packages/calling/calling/src/CallingClient/CallingClient.test.ts:1419]: we have logSpy globally present too, we don't need to spy again and this was causing eslint error

File: .eslintignore [hunk 11]
```diff
@@ -37,3 +37,4 @@ umd/
 types/
 webex.min.js
 *.yml
+packages/calling
```
[the remark this answers is not in the document]
Comment (Kesari3008) [reply .eslintignore:40]: we are only ignoring the global .eslintrc.js rule to be applied on calling folder. But eslintrc file inside calling folder checks for eslint errors and warnings

File: packages/calling/calling/jest.expectExtensions.js (hunk 1 above)
Comment (Kesari3008) [reply packages/calling/calling/jest.expectExtensions.js:5]: done

File: packages/calling/calling/jest.expectExtensions.js (hunk 2 above)
Comment (Kesari3008) [reply packages/calling/calling/jest.expectExtensions.js:14]: done

File: packages/calling/calling/versionPlugin.js (hunk 4 above)
Comment (Kesari3008) [reply packages/calling/calling/versionPlugin.js:33]: done

File: packages/calling/calling/versionPlugin.js (hunk 5 above)
Comment (Kesari3008) [reply packages/calling/calling/versionPlugin.js:11]: done

File: .eslintignore (hunk 11 above)
[the remark this answers is not in the document]
Comment (Kesari3008) [reply .eslintignore]: we only have global eslint present now

File: packages/calling/calling/.editorconfig [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (mkesavan13) [inline packages/calling/calling/.editorconfig:13]: Should we have the editorconfig out of the package?

We also already have a VS Code config file -> https://github.com/webex/webex-js-sdk/blob/master/.vscode/settings.json

File: webpack.config.js [hunk 13]
```diff
@@ -20,13 +20,15 @@ dotenv.config({path: '.env.default'});
...
+  entry: {
+    webex:
+      env && env.NODE_ENV === 'development'
+        ? `${path.resolve(__dirname)}/packages/webex/src/index.js`
+        : './packages/webex',
+  },
   mode: env && env.NODE_ENV === 'development' ? 'development' : 'production',
   output: {
-    filename: 'webex.min.js',
+    filename: '[name].min.js',
```
Comment (mkesavan13) [inline webpack.config.js:31]: What is the purpose of this change?

**Review by mkesavan13 — COMMENTED** [review]
Two common questions

- Why do we have the calling folder inside `packages/calling` instead of just `packages/`?
- The documentation files are to be pushed by the CI itself by running a doc command after each merge. This setup is already available for the Web SDK. The doc files seem to be a lot in number causing a spike in the change set. Can we remove them from the PR?

We should at least have the docs generated and pushed to a folder under [the docs folder](https://github.com/webex/webex-js-sdk/tree/master/docs) say `docs/api/calling`.  This way, it will be accessible from this URL -> https://webex.github.io/webex-js-sdk/api/calling/

**Comment by mkesavan13** [comment]
Also @arun3528 

Wasn't it the plan to include the `callings` folder also inside the `webex` folder as we did for `meetings`?

And could you please pull the latest master changes into the calling branch? That would help @Kesari3008 in setting up the webpack config and stuff I think.

**Comment by Kesari3008** [comment]
Closing this PR since using rollup is not possible. New PR raised to take care of webex initialization.

**Comment by Kesari3008** [comment]
New PR raised: https://github.com/webex/webex-js-sdk/pull/2908 to take care of webex initialization. Thse changes cannot be merged since we cannot use rollup
